src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/network/access/LocalPublicIpv4Provider.kt

```diff
@@ -0,0 +1,61 @@
+package com.atlassian.performance.tools.awsinfrastructure.api.network.access
+
+import com.atlassian.performance.tools.awsinfrastructure.Ipv4Validator
+import com.atlassian.performance.tools.awsinfrastructure.VerboseUrlReader
+import java.net.URL
+import java.util.function.Supplier
+
+class LocalPublicIpv4Provider private constructor(
+    private val servicesToQuery: List<URL>,
+    private val urlReader: (serviceUrl: URL) -> String?,
+    private val ipValidator: (ip: String) -> Boolean
+) : () -> String, Supplier<String> {
+    object Defaults {
+        val servicesToQuery: List<URL> = listOf(
+            URL("https://checkip.amazonaws.com")
+        )
+        val urlReader: (serviceUrl: URL) -> String? = VerboseUrlReader()
+        val ipValidator: (ip: String) -> Boolean = Ipv4Validator()
+    }
+
+    constructor() : this(
```

Comment:
We've got a builder with defaults already. What's the use case for the extra constructor? IDE autocompletion?

Reply:
I used it to avoid need of writing `LocalPublicIpv4Provider.Builder().build()` and instead just be able to `LocalPublicIpv4Provider()`. Do you think it's not worth it and I should remove this constructor and always use the `Builder`?

Reply:
It's ok. Empathizing with callers is a good thing :slightly_smiling_face: